akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/common/ElasticDAO.scala

```diff
@@ -0,0 +1,174 @@
+import org.elasticsearch.action.support.WriteRequest.RefreshPolicy
+import org.elasticsearch.common.settings.Settings
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+object ElasticDAO {
+
+  case class SaveInIndex(index: String, typeName: String, jsonDocs: Seq[ElasticDocumentInsertable])
+
+  case class ErrorSavingData(
+                              error: Exception,
+                              index: String,
+                              typeName: String,
+                              jsonDocs: Seq[ElasticDocumentInsertable]
+                            )
+
+  case class DataSavedConfirmation(index: String, typeName: String, jsonDocs: Seq[ElasticDocumentInsertable])
+
+  case class DeleteFromIndex(
+                              index: String,
+                              typeName: String,
+                              where: Option[QueryDefinition] = None
+                            )
+
+  case class ErrorDeletingData(
+                              error: Exception,
+                              index: String,
+                              typeName: String,
+                              originalRQ: DeleteFromIndex
+                              )
+
+  case class DeleteConfirmation(index: String, typeName: String, originalRQ: DeleteFromIndex)
+
+  case class RetrieveAllFromIndexSorted(
+                                         index: String,
+                                         typeName: String,
+                                         where: Option[QueryDefinition] = None,
+                                         sort: Option[FieldSortDefinition] = None,
+                                         limit: Option[Int] = None)
+
+  case class DataRetrieved(
+                            index: String,
+                            typeName: String,
+                            data: RichSearchResponse,
+                            originalRQ: RetrieveAllFromIndexSorted
+                          )
+
+  case class ErrorRetrievingData(
+                                  error: Exception,
+                                  index: String,
+                                  typeName: String,
+                                  originalRQ: RetrieveAllFromIndexSorted
+                                )
+
+  object InitElasticDAO
+  case class ProblemConnectingWithES(e: Exception)
+  object ReadyToListen
+
+  private val settings = Settings.builder()
+    .put("cluster.name", TapiporlaConfig.ElasticSearch.clusterName)
+    .put("xpack.security.user", TapiporlaConfig.ElasticSearch.authUser)
+    .build()
+
+  val client = XPackElasticClient(settings, ElasticsearchClientUri(TapiporlaConfig.ElasticSearch.endpoint))
+
+}
+
+
+trait ElasticDAO extends Actor with ActorLogging with Stash{
+
+  def indexCreation: CreateIndexDefinition //Index to be created (if already exists we don't check if the mappings are similar)
+
+  self ! InitElasticDAO
```

Comment:
I'm more fan of doing this stuff, not in trait body but in ```preStart``` method.
This other way, the behaviour can be overridden.

Reply:
Yes, you are correct. It was my first time autosending me on init messages. Sorry. Changed